### PR TITLE
Add insert_with_values to ListStore and TreeStore

### PR DESCRIPTION
**Pull request by gkoz**

### CODE REVIEW

File: src/list_store.rs [hunk 1]
```diff
@@ -21,6 +21,30 @@ impl ListStore {
         }
     }
 
+    pub fn insert_with_values(&self, position: Option<u32>, columns: &[u32], values: &[&ToValue])
+            -> TreeIter {
+        unsafe {
+            assert!(position.unwrap_or(0) <= i32::max_value() as u32);
+            let n_columns = ffi::gtk_tree_model_get_n_columns(self.to_glib_none().0) as u32;
+            assert!(columns.len() == values.len());
```
Comment (GuillaumeGomez) [inline src/list_store.rs]: Shouldn't this `assert!` be done before the previous line?

File: src/list_store.rs [hunk 2]
```diff
@@ -21,6 +21,30 @@ impl ListStore {
...
+            assert!(columns.len() <= n_columns as usize);
+            for &column in columns {
+                assert!(column < n_columns);
+                let type_ = from_glib(
+                    ffi::gtk_tree_model_get_column_type(self.to_glib_none().0, column as c_int));
+                assert!(Value::type_transformable(values[column as usize].to_value_type(), type_));
```
Comment (GuillaumeGomez) [inline src/list_store.rs]: Wasn't it supposed to be `assert_eq`?

File: src/tree_store.rs [hunk 3]
```diff
@@ -20,6 +20,31 @@ impl TreeStore {
...
+        unsafe {
+            assert!(position.unwrap_or(0) <= i32::max_value() as u32);
+            let n_columns = ffi::gtk_tree_model_get_n_columns(self.to_glib_none().0) as u32;
+            assert!(columns.len() == values.len());
+            assert!(columns.len() <= n_columns as usize);
+            for &column in columns {
+                assert!(column < n_columns);
+                let type_ = from_glib(
+                    ffi::gtk_tree_model_get_column_type(self.to_glib_none().0, column as c_int));
+                assert!(Value::type_transformable(values[column as usize].to_value_type(), type_));
```
Comment (GuillaumeGomez) [inline src/tree_store.rs]: `assert_eq`?

File: src/tree_store.rs [hunk 4]
```diff
@@ -20,6 +20,31 @@ impl TreeStore {
         }
     }
 
+    pub fn insert_with_values(&self, parent: &TreeIter, position: Option<u32>, columns: &[u32],
+            values: &[&ToValue]) -> TreeIter {
+        unsafe {
+            assert!(position.unwrap_or(0) <= i32::max_value() as u32);
+            let n_columns = ffi::gtk_tree_model_get_n_columns(self.to_glib_none().0) as u32;
+            assert!(columns.len() == values.len());
```
Comment (GuillaumeGomez) [inline src/tree_store.rs]: Should be swapped with previous line.

File: src/list_store.rs (hunk 2 above)
Comment (gkoz) [inline src/list_store.rs]: `assert_eq!` is much more expensive than `assert!`, I wouldn't use it outside of tests.

File: src/list_store.rs (hunk 1 above)
Comment (gkoz) [inline src/list_store.rs]: Yeah I could move them around.

File: src/list_store.rs (hunk 2 above)
Comment (gkoz) [inline src/list_store.rs]: Oh wait, I missed the point. The correct answer is: of course not!

**Comment by GuillaumeGomez** [comment]
Please squash and we're good to go!

**Comment by gkoz** [comment]
Done

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**